server/server.js

```diff
@@ -1,33 +1,87 @@
+require('lib/logger');
+
 import { resolve } from 'path';
+import semver from 'semver';
 import Koa from 'koa';
 import bodyParser from 'koa-better-body';
 import favicon from 'koa-favicon';
 import convert from 'koa-convert';
 import views from 'koa-views';
-import config from './config';
+import mount from 'koa-mount';
+import serve from 'koa-static';
+
 import store from './middleware/store';
 import render from './middleware/render';
 import routes from './routes';
+import utils from '../lib/utils';
 
-const app = new Koa();
+if (semver.lt(process.version, '7.6.0')) {
+  console.error('Node Version should be greater than 7.6.0');
+  process.exit(-1);
+}
+
+const config = utils.getServerConfig();
 
-// Serve static files
-if (process.env.NODE_ENV !== 'production') {
-  const mount = require('koa-mount');
-  const serve = require('koa-static');
+global.HOSTNAME = config.http.hostname || 'localhost';
+global.PORT = config.http.port || 8000;
 
-  for (const [k, v] of Object.entries(config.http.static)) {
-    app.use(mount(k, serve(v, { index: false })));
+const app = new Koa();
+
+// serve static files
+const serveFiles = function(env){
+  for (const [k, v] of Object.entries(config.http.static[env])) {
+    app.use(mount(k, serve(utils.root(v), { index: false })));
   }
+};
+
+if (process.env.NODE_ENV === 'development') {
+  // disable babel server env plugins transform
+  process.env.BABEL_ENV = '';
+
+  const webpack = require('webpack');
+  const webpackMiddleware = require('koa-webpack');
+  const webpackConfig = require('../webpack.dev');
+
+  app.use(webpackMiddleware({
+    compiler: webpack(webpackConfig),
+    dev: {
+      publicPath: webpackConfig.output.publicPath,
+      noInfo: false,
+      quiet: false,
+      watchOptions: {
+        aggregateTimeout: 300,
+        poll: false,
+        ignored: /node_modules/,
+      },
+      stats: {
+        colors: true,
+        // hash: false,
+        // timings: false,
+        // version: false,
+        // chunks: false,
+        modules: true,
+        // children: false,
+        chunkModules: true,
+      },
+    },
+  }));
+
+  serveFiles('development');
+}
+
+
+if (process.env.NODE_ENV === 'production') {
+  serveFiles('production');
```

Comment:
simplify these codes to ``serveFiles(process.env.NODE_ENV || 'development')``

Reply:
ok, will refactor it in next commit